src/native/libs/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -678,9 +679,16 @@ int CryptoNative_SslCtxSetCaching(SSL_CTX* ctx, int mode, int cacheSize, SslCtxN
     {
         SSL_CTX_set_options(ctx, SSL_OP_NO_TICKET);
     }
-    else if (cacheSize >= 0)
+    else
     {
-        SSL_CTX_ctrl(ctx, SSL_CTRL_SET_SESS_CACHE_SIZE, (long)cacheSize, NULL);
+        unsigned char id[SSL_MAX_SID_CTX_LENGTH];
+        getrandom(id, sizeof(id), GRND_RANDOM);
+        SSL_CTX_set_session_id_context(ctx, id, sizeof(id));
```

Comment:
Don't use getrandom.

```suggestion
        uint8_t id[SSL_MAX_SID_CTX_LENGTH];
        
        if (!RAND_bytes(id, sizeof(id)))
        {
            // No idea what you should do here.  Maybe just use the RNG.fill in managed and pass a session ID down.
        }

        SSL_CTX_set_session_id_context(ctx, id, sizeof(id));
```

Reply:
I was thinking about passing it from managed. So we would have visibility even if only for debug. Is RNG.fill all pseudo random computed in managed code? I don't think we need strong crypto entropy here - it is just another layer to isolate sensation and find the one that match.

Reply:
> Is RNG.fill all pseudo random computed in managed code?

The session IDs should either be entirely predictable (incremental, derived, etc) or entirely non-predictable, not from a PRNG that isn't a CSPRNG.  "Bad random" doesn't belong in security protocols.

I believe nginx uses a derived value (a hash of the client cert thumbprint and the thumbprints of any certs sent in the CA trust list?).  OpenSSL s_server uses the CSPRNG.  I think SChannel uses a partially deterministic/partially CSPRNG value (such as random guids always have that fixed '4') (I might be conflating two different "session ID" values, so it might be pure CSPRNG).

Reply:
so `RNG.Fill` or `RAND_bytes`?

Reply:
They're the same.  Calling RandomNumberGenerator.Fill in managed code, then passing the session ID in, makes the exception model simpler.